src/xscen/ensembles.py

```diff
@@ -629,13 +631,150 @@ def generate_weights(  # noqa: C901
     return weights
 
 
+def _partition_from_list(datasets, partition_dim, subset_kw, regrid_kw):
+    list_ds = []
+    # only keep attrs common to all datasets
+    common_attrs = False
+    for ds in datasets:
+        if subset_kw:
+            ds = subset(ds, **subset_kw)
+            ds = ds.drop_vars(
+                ["lat", "lon", "rlat", "rlon", "rotated_pole"], errors="ignore"
+            )
```

Comment:
I'd use a more generic approach.

```suggestion
            gridmap = xs.spatial.get_grid_mapping(ds)
            ds = ds.drop_vars(
                [ds.cf["longitude"], ds.cf["latitude"], ds.cf.axes["X"][0], ds.cf.axes["Y"][0], gridmap], errors="ignore"
            )
```

Reply:
this doesnt work exactly, but i've added a workign version